engine/src/tilecachegl3.x.cpp

```diff
@@ -288,7 +522,12 @@ bool MCTileCacheOpenGLCompositor_EndTiling(void *p_context)
 {
 	MCTileCacheOpenGLCompositorContext *self;
 	self = (MCTileCacheOpenGLCompositorContext *)p_context;
-	
+
+#if defined(MCANDROIDUSEHARDWAREBUFFER)
+	// Unlock any locked supertiles
+	MCTileCacheOpenGLCompositorUnlockSuperTiles(self);
+#endif
+
 	// Flush any empty textures.
 	MCTileCacheOpenGLCompositorFlushSuperTiles(self, false);
```

Comment:
It looks like we can elide HardwareBuffer unlock into Flush - i.e. just have:
```
MCTileCacheHardwareBufferUnlock(t_super_tile)
```
Called on each super-tile which exists.

Reply:
Fixed in updated commits